### PR TITLE
Reduced anonymous functions, refactored error handing, simplified cod…

### LINKED ISSUE #28: Improve code quality by reducing anonymous functions
For better code quality and readability we need to reduce amount of anonymous functions that could be replaced as regular functions. Most prior anon functions usage to reduce:

- [ ] Remove fn from `rpcCallCreate` and retrieve post functions as named ones
- [ ] Remove fnAfter from `discoverSChainNetwork`
- [ ] Retrieve anon function from fnSignMsgOracle

### PR DESCRIPTION
**Pull request by sergiy-skalelabs**
Reduced anonymous functions.
Refactored error handing.
Simplified code structure.
There are no parallel duplicated log writing.
No performance changes introduced.
No new tests needed.

### CODE REVIEW

File: src/bls.mjs [hunk 1]
```diff
@@ -2500,54 +2381,38 @@ export async function handleSkaleImaBSU256( joCallData ) {
...
+        const joIn = {
+            "jsonrpc": "2.0",
+            "id": randomCallID(),
+            "method": "blsSignMessageHash",
+            "params": {
+                "keyShareName": optsBSU256.joAccount.strBlsKeyName,
+                "messageHash": optsBSU256.strMessageHash,
+                "n": optsBSU256.nParticipants,
+                "t": optsBSU256.nThreshold,
+                "signerIndex": signerIndex + 0 // 1-based
```
Comment (dmytrotkk) [inline src/bls.mjs]: Why adding `0` here?

File: src/bls.mjs [hunk 2]
```diff
@@ -1276,27 +1259,10 @@ async function doSignProcessOneImpl( i, optsSignOperation ) {
...
-                    "{p}JSON RPC call(doSignProcessOneImpl) to S-Chain node {} failed, " +
-                    "RPC call was not created, error is: {err}, sequence ID is {}",
-                    optsSignOperation.strLogPrefix, strNodeDescColorized, err,
-                    optsSignOperation.sequenceId );
-                if( joCall )
-                    await joCall.disconnect();
-                return;
-            }
+    let joCall = null;
+    rpcCall.create( strNodeURL, rpcCallOpts )
```
Comment (dmytrotkk) [inline src/bls.mjs]: can be changed to 

```javascript
const joCall = await rpcCall.create( strNodeURL, rpcCallOpts );
// the rest of the logic here
```

**Review by dmytrotkk — CHANGES_REQUESTED** [review]

File: src/bls.mjs (hunk 1 above)
Comment (sergiy-skalelabs) [reply src/bls.mjs]: `const signerIndex = optsBSU256.imaState.nNodeNumber;` ... adding zero just guarantees no object reference
but here I will remove it

Comment (sergiy-skalelabs) [reply src/bls.mjs]: removed

File: src/bls.mjs (hunk 2 above)
Comment (sergiy-skalelabs) [reply src/bls.mjs]: Done

File: src/pwa.mjs [hunk 3]
```diff
@@ -236,15 +236,11 @@ async function notifyOnLoopImpl( imaState, strLoopWorkType, nIndexS2S, isStart )
...
-                strNodeURL, rpcCallOpts, async function( joCall, err ) {
-                    if( err ) {
-                        log.error(
-                            "PWA failed to create loop-{} notification RPC call to node #{} with " +
-                            "URL {url}, error is: {err}", se, i, strNodeURL, err );
-                        return;
-                    }
-                    joCall.call( { // NOTICE: no await here, executed async
+            let joCall = null;
+            rpcCall.create( strNodeURL, rpcCallOpts ) // NOTICE: no await here, executed async
```
Comment (DmytroNazarenko) [inline src/pwa.mjs]: Can be await

**Review by dmytrotkk — APPROVED** [review]

**Review by DmytroNazarenko — APPROVED** [review]

File: src/pwa.mjs [hunk 4]
```diff
@@ -236,39 +236,34 @@ async function notifyOnLoopImpl( imaState, strLoopWorkType, nIndexS2S, isStart )
...
-                                "PWA failed to perform loop-{} notification RPC call to node #{} " +
-                                "with URL {url}, error is: {err}", se, i, strNodeURL, err );
-                            await joCall.disconnect();
-                            return;
-                        }
-                        if( imaState.isPrintPWA ) {
-                            log.success( "Was successfully sent PWA loop-{} notification to " +
-                                "node #{} with URL {url}", se, i, strNodeURL );
-                        }
+            let joCall = rpcCall.create( strNodeURL, rpcCallOpts )
```
Comment (DmytroNazarenko) [inline src/pwa.mjs]: Should be await

**Review by DmytroNazarenko — CHANGES_REQUESTED** [review]

File: src/pwa.mjs (hunk 4 above)
Comment (sergiy-skalelabs) [reply src/pwa.mjs]: fixed

File: src/pwa.mjs (hunk 3 above)
Comment (sergiy-skalelabs) [reply src/pwa.mjs]: outdated

**Review by DmytroNazarenko — APPROVED** [review]

**Review by DmytroNazarenko — APPROVED** [review]

**Review by dmytrotkk — APPROVED** [review]

**Pull request merged**